cli/api/daemon.go

```diff
@@ -841,10 +841,21 @@ func (d *daemon) startAgent() error {
 				}
 				err = masterClient.UpdateHost(updatedHost)
 				if err != nil {
-					log.WithError(err).Warn("Unable to update master with delegate host information")
-					return "" // Try again
+					log.WithError(err).Warn("Unable to update master with delegate host information. Retrying silently")
+					go func(masterClient *master.Client, updatedHost host.Host) {
+						err := errors.New("")
+						for err != nil {
+							select {
+								case <-d.shutdown:
+									return
+								default:
+									time.Sleep(5)
```

Comment:
I think you want `time.Sleep(5 * time.Second)`; otherwise it's only sleeping 5 nanoseconds